src/lib/expression/evaluation/expression_evaluator.cpp

```diff
@@ -547,7 +547,10 @@ ExpressionEvaluator::_evaluate_predicate_expression<ExpressionEvaluator::Bool>(
       return _evaluate_binary_predicate_expression<ExpressionEvaluator::Bool>(
           static_cast<const BinaryPredicateExpression&>(predicate_expression));
 
-    case PredicateCondition::Between:
+    case PredicateCondition::BetweenInclusive:
```

Comment:
This will not work, how are all four Between-variant to be handled by the same code?!? Please also add tests for this to the ExpressionEvaluatorTests, should be straight forward.

Reply:
Let me know if you need help adjusting the ExpressionEvaluator.

Reply:
should be fixed